wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,344 @@
+        reason = ''.join([i+' ' for i in args if not re.match(r'<@!?[0-9]*>', i)])
+        reason = reason if reason else "No Reason Given!"
+        logger.info(f"[Ban ban()] reason for ban(s)={reason}")
+
+        # ban
+        dm = True
+        for user in users_to_ban:
+            username = user.name + '#' + user.discriminator
+            logger.info(f"[Ban ban()] Banning {username} with id {user.id}")
+
+            # add to ban_list
+            self.ban_list.append(user.id)
+
+            # dm banned user
+            e_obj = discord.Embed(title="Ban Notification",
+                                  description="**You've been PERMANENTLY BANNED from" +
+                                              f"{self.bot.guilds[0].name.upper()}.**",
+                                  color=discord.Color.red(),
+                                  timestamp=datetime.datetime.now(pytz.timezone('Canada/Pacific')))
+
+            e_obj.add_field(name='Reason',
+                            value=f"```{reason}```\n" +
+                            "Please refrain from this kind of behaviour in the future. Thank you.")
+
+            e_obj.set_footer(icon_url=self.bot.guilds[0].icon_url, text=self.bot.guilds[0])
+            try:
+                if e_obj:
+                    await user.send(embed=e_obj)
+                    logger.info("[Ban ban()] User notified via dm of their ban")
+            except (discord.HTTPException, discord.Forbidden, discord.InvalidArgument):
+                dm = False
+                logger.info("[Ban ban()] Notification dm to user failed due to user preferences")
+
+            # kick
+            await user.kick(reason=reason)
+            dt = datetime.datetime.now(pytz.utc)
+            logger.info(f"[Ban ban()] User kicked from guiled at {dt}.")
+
+            # report to council
+            e_obj = discord.Embed(title="Ban Hammer Deployed",
+                                  colour=discord.Color.red())
+
+            e_obj.add_field(name="Banned User", value=f"**{username}**", inline=True)
+            e_obj.add_field(name="Moderator", value=f"**{mod_info[0]}**", inline=True)
+            e_obj.add_field(name="Reason", value=f"```{reason}```", inline=False)
+            e_obj.add_field(name="Notification DM", value="SENT\n" if dm else "NOT SENT, DUE TO USER DM PREF's\n",
+                            inline=False)
+            e_obj.set_footer(text="Moderator Action")
+            if e_obj:
+                e_obj.timestamp = dt.astimezone(pytz.timezone('Canada/Pacific'))
+                await self.mod_channel.send(embed=e_obj)
+            logger.info(f"[Ban ban()] Message sent to mod channel,{self.mod_channel}, of the ban for {username}.")
+
+            # update db
+            await BannedUsers.insert_ban(username, user.id)
+            await BanRecords.insert_record(username, user.id, mod_info[0], mod_info[1], dt, reason)
+
+    @commands.command()
+    async def unban(self, ctx, _id: int):
```

Comment:
why call id `_id`?

Reply:
when i called it `id` it was conflicting with something and i didnt care to looking into more. its just var name so i didnt care.

Reply:
please look into it.

Reply:
found it: https://stackoverflow.com/questions/6350847

name of a built-in function